sdk/eventhub/test-resources.json

```diff
@@ -175,12 +184,23 @@
         "deleteRetentionPolicy": {
           "enabled": false
         }
-      }
+      },
+      "resources": [
+        {
+          "name": "azure-webjobs-eventhub",
+          "type": "containers",
+          "apiVersion": "2019-04-01",
+          "dependsOn": [
+            "[resourceId('Microsoft.Storage/storageAccounts/blobServices', variables('storageAccount'), 'default')]"
+          ],
+          "properties": {}
+        }
+      ]
     },
     {
       "type": "Microsoft.Authorization/roleAssignments",
       "apiVersion": "2019-04-01-preview",
-      "name": "[guid(resourceGroup().id, deployment().name, parameters('baseName'), variables('eventHubsDataOwnerRoleId'))]",
+      "name": "[guid(resourceGroup().id, deployment().name, parameters('baseName'), variables('eventHubsDataOwnerRoleId'), parameters('testApplicationOid'))]",
```

Comment:
Including `testApplicationOid` into as a factor into this guids randomness makes it so re-deploying with a different service principal doesn't fail.